File: Data Structures & Algorithms/three-integer-sum/submission-9.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        nums.sort();
        sol = [];
        for i, a in enumerate(nums):
            if(a > 0):
                break;
            if i > 0 and nums[i] == nums[i-1]:
                continue
            l = i+1;
            r = len(nums)-1;
            while(r > l):
                if(a + nums[l] + nums[r] < 0):
                    l += 1;
                elif(a + nums[l] + nums[r] > 0):
                    r -= 1;
                else:
                    sol.append([a, nums[l], nums[r]]);
                    l += 1;
                    while nums[l - 1] == nums[l] and r > l:
                        l += 1;
        return sol;




                    

File: Data Structures & Algorithms/top-k-elements-in-list/submission-9.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
       HashMap<Integer, ArrayList<Integer>> map = new HashMap<>();
       int[] sol = new int[k];
       for(int num : nums) {
        if(map.containsKey(num)) {
            map.get(num).add(num);
        } else {
            ArrayList<Integer> list = new ArrayList<>();
            list.add(num);
            map.put(num, list);
        }
       } 
       
       for(int i = 0; i < k; i++) {
        int max = 0;
        int val = 0;
        for(Map.Entry<Integer, ArrayList<Integer>> entry : map.entrySet()) {
            if(entry.getValue().size() > max) {
                max = entry.getValue().size();
                val = entry.getKey();
            } 
        }
        map.remove(val);
        sol[i] = val;
       }
       return sol;
    }
}
